File: WebClient/FleetBrotherWebClient/src/app/data/dto/alert.dto.ts
export interface AlertBase {
    id: number,
    name: string,
    keyName: string,
}

export interface MinValueAlert extends AlertBase{
    minValue: number,
}

export interface MaxValueAlert extends AlertBase{
    maxValue: number,
}

export interface ForbiddenValueAlert extends AlertBase{
    forbiddenValue: number | boolean | string,
}

export interface ExistsValueAlert extends AlertBase{
    exists: boolean,
}
export type Alert = MinValueAlert | MaxValueAlert | ForbiddenValueAlert | ExistsValueAlert;

File: Server/FleetBrotherServer/src/main/kotlin/hu/bme/aut/fleetbrotherserver/dtos/AlertDto.kt
package hu.bme.aut.fleetbrotherserver.dtos

import java.io.Serializable

/**
 * DTO for {@link hu.bme.aut.fleetbrotherserver.data.entities.Alert}
 */
data class AlertDto(
    val id: Int? = null,
    val name: String,
    val keyName: String,
    val minValue: Double? = null,
    val maxValue: Double? = null,
    val forbiddenValue: String? = null,
    val exists: Boolean? = null,
    val isDeleted: Boolean? = false
) : Serializable